#
# old_revision [79edc903796100d175f9068b21615b765d13802c]
#
# patch "mtn-browse"
#  from [693ee97c8c3337fb3389121a69c942715f495837]
#    to [960aec82136330eab2da350a8b70992190b7dd9d]
#
============================================================
--- mtn-browse	693ee97c8c3337fb3389121a69c942715f495837
+++ mtn-browse	960aec82136330eab2da350a8b70992190b7dd9d
@@ -79,7 +79,7 @@ use Gtk2::Pango;
 use Gtk2::GladeXML;
 use Gtk2::Helper;
 use Gtk2::Pango;
-use Gtk2::SourceView;
+use Gtk2::SourceView2;
 use IO::Dir;
 use IO::File;
 use IPC::Open3;
@@ -2402,13 +2402,13 @@ sub get_browser_window(;$$$$$)
 
         # Setup the file file viewer (with syntax highlighting).
 
-        $browser->{file_view_svbuffer} = Gtk2::SourceView::Buffer->new(undef);
-        $browser->{file_view_svbuffer}->set_check_brackets(FALSE);
+        $browser->{file_view_svbuffer} = Gtk2::SourceView2::Buffer->new(undef);
+        $browser->{file_view_svbuffer}->set_highlight_matching_brackets(FALSE);
         $browser->{file_view_svbuffer}->set_max_undo_levels(0);
         $browser->{file_view_svbuffer}->begin_not_undoable_action();
         $browser->{file_view_svlangmgr} =
-            Gtk2::SourceView::LanguagesManager->new();
-        $browser->{file_view_sv} = Gtk2::SourceView::View->
+            Gtk2::SourceView2::LanguageManager->new();
+        $browser->{file_view_sv} = Gtk2::SourceView2::View->
             new_with_buffer($browser->{file_view_svbuffer});
         $browser->{file_view_sv}->modify_font($mono_font);
         $browser->{file_view_sv}->set_cursor_visible(FALSE);
@@ -3505,7 +3505,7 @@ sub update_browser_state($$)
             $browser->{file_view_svbuffer}->
                 place_cursor($browser->{file_view_svbuffer}->get_start_iter());
             $browser->{file_view_svbuffer}->set_text("");
-            $browser->{file_view_svbuffer}->set("highlight", FALSE);
+            $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
             set_label_value($browser->{file_name_value_label}, "");
             set_label_value($browser->{file_author_value_label}, "");
             set_label_value($browser->{last_update_value_label}, "");
@@ -3631,7 +3631,7 @@ sub display_file($$)
     $browser->{file_view_svbuffer}->
         place_cursor($browser->{file_view_svbuffer}->get_start_iter());
     $browser->{file_view_svbuffer}->set_text("");
-    $browser->{file_view_svbuffer}->set("highlight", FALSE);
+    $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
 
     # Get contents.
 
@@ -3735,9 +3735,12 @@ sub display_file($$)
                          && $mime_details->{syntax_highlight}))
                     && defined($lang =
                                $browser->{file_view_svlangmgr}->
-                                   get_language_from_mime_type($mime_type)))
+                                   guess_language($browser->
+                                                      {file_being_viewed}->
+                                                      {short_name},
+                                                  $mime_type)))
                 {
-                    $browser->{file_view_svbuffer}->set("highlight", TRUE);
+                    $browser->{file_view_svbuffer}->set_highlight_syntax(TRUE);
                     $browser->{file_view_svbuffer}->set_language($lang);
                 }
 
